explorer/src/lib/components/block-details/BlockDetails.svelte

```diff
@@ -0,0 +1,170 @@
+<svelte:options immutable={true} />
+
+<script>
+  import { mdiArrowLeft, mdiArrowRight } from "@mdi/js";
+  import { AppAnchor, ListItem } from "$lib/components";
+  import { Card, Icon, ProgressBar, Switch } from "$lib/dusk/components";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import {
+    calculateAdaptiveCharCount,
+    getRelativeTimeString,
+    middleEllipsis,
+  } from "$lib/dusk/string";
+  import { onMount } from "svelte";
+  import "./BlockDetails.css";
+
+  const formatter = createCurrencyFormatter("en", "DUSK", 0);
+  const blockHeight = 34526;
+
+  /** @type {number} */
+  let screenWidth = window.innerWidth;
+
+  /** @type {boolean} */
+  let togglePayload = false;
+
+  onMount(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      const entry = entries[0];
+
+      screenWidth = entry.contentRect.width;
+    });
+
+    resizeObserver.observe(
+      document.body.getElementsByClassName("details-list__definition")[0]
+    );
+
+    return () => resizeObserver.disconnect();
+  });
+</script>
+
+<Card className="block-details">
+  <header slot="header" class="block-details__header">
+    <h3 class="block-details__header-heading">
+      Block Details <span class="block-details__header-heading--block-height">
+        - #{formatter(blockHeight)}</span
```

Comment:
I don't think this should be the currency formatter, although it will achieve the same.

This is what we have for the blocks on the web-wallet:

<img width="576" alt="Screenshot 2024-04-13 at 14 58 41" src="https://github.com/dusk-network/rusk/assets/34094428/56f0e6c9-7e14-4afa-bc16-a73c87f8f743">

Maybe we also need an integer number formatter.

Reply:
Why not, if they have the same outcome?

Reply:
Also why are you reviewing this on a saturday?

Reply:
> Why not, if they have the same outcome?

Mostly for semantic reasons. Here, the `formatter` is a `CurrencyFormatter`, and the block is just an integer. Separation of concerns is also a practice good to follow, the currency formatter might change as it can unexpectedly mess up the formatting of values, which are not currency-related.

Reply:
I'm created a separate formatter just for Number so that we separate concerns